apps/glusterfs/placer_arbiter.go

```diff
@@ -371,13 +371,24 @@ func (dscan *arbiterDeviceScanner) Scan(index int) <-chan string {
 	return dscan.dataDevs
 }
 
-func discountBrickSize(dataBrickSize, averageFileSize uint64) (uint64, error) {
+func discountBrickSize(dataBrickSize, averageFileSize uint64) (brickSize uint64,
+	err error) {
+
 	if dataBrickSize < averageFileSize {
 		return 0, fmt.Errorf(
 			"Average file size (%v) is greater than Brick size (%v)",
 			averageFileSize, dataBrickSize)
 	}
-	return (dataBrickSize / averageFileSize), nil
+
+	brickSize = dataBrickSize / averageFileSize
+
+	if brickSize < 16*MB {
```

Comment:
I would have loved a comment that this is the minimal device size for XFS. Oh well, its in the commit message at least :-)

Reply:
Good point. Updated the patch to a more verbose log message instead of adding a comment. This way, it is even apparent from the log why this happened...